New exception as of 0.6.16

I confirmed this issue is new as of `v0.6.16`:

```
panic: value is not known

goroutine 17 [running, locked to thread]:
github.com/zclconf/go-cty/cty.Value.LengthInt({{{0x116efa150?, 0x117dec2e0?}}, {0x116bbcd40?, 0x117dbb8e0?}})
        /var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/pkg/mod/github.com/zclconf/go-cty@v1.16.3/cty/value_ops.go:1158 +0x268
github.com/zclconf/go-cty/cty.Value.AsValueSlice({{{0x116efa150?, 0x117dec2e0?}}, {0x116bbcd40?, 0x117dbb8e0?}})
        /var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/pkg/mod/github.com/zclconf/go-cty@v1.16.3/cty/value_ops.go:1451 +0x6c
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.getChildBlocks.func1(...)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:583
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.getChildBlocks(0x116c4a500?)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:592 +0x3e4
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.(*terraformConverter).buildBlock(0x14000306000, 0x14000a43c00)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:216 +0xac
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.(*terraformConverter).visitBlock(0x14000306000, 0x14000a43c00, {0x1400030bcc8, 0x15}, 0x14000a50360)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:148 +0x1ec
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.(*terraformConverter).visitModule(0x14000306000, 0x140005ff0e0, 0x14000a50360)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:136 +0x6c
github.com/cloud-custodian/tfparse/gotfparse/pkg/converter.(*terraformConverter).VisitJSON(0x14000306000)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/pkg/converter/converter.go:121 +0x98
main.Parse(0x0?, 0x0, 0x0, 0x0, 0x600003d18bb0, 0x0, 0x16b199270)
        /private/var/folders/w5/_8wgjw3j5cg6mgrth3s2kg9m0000gn/T/tmp903kmsyo/src/github.com/cloud-custodian/tfparse/gotfparse/gotfparse/cmd/tfparse/main.go:50 +0x534
```

My gut is that it's something upstream similar to https://github.com/cloud-custodian/tfparse/issues/228 but still looking into it

--- a/gotfparse/pkg/converter/converter.go
+++ b/gotfparse/pkg/converter/converter.go
@@ -576,7 +576,7 @@ func getChildBlocks(b *terraform.Block) []*terraform.Block {
 		attr := b.GetAttribute("for_each")
 
 		value := attr.Value()
-		if value.IsNull() {
+		if value.IsNull() || !value.IsKnown() {
 			return 0
 		}
 

--- a/tests/terraform/dynamic-stuff/main.tf
+++ b/tests/terraform/dynamic-stuff/main.tf
@@ -38,4 +38,16 @@ resource "some_resource" "this" {
   }
 
   prop3 = "end"
+
+  dynamic "unknown_loop" {
+    for_each = var.unknown
+
+    content {
+      other = each.value
+    }
+  }
+}
+
+variable "unknown" {
+  type = set(string)
 }

--- a/tests/test_tfparse.py
+++ b/tests/test_tfparse.py
@@ -280,11 +280,25 @@ def test_parse_dynamic_content(tmp_path):
     # mod_path = init_module("dynamic-stuff", tmp_path)
     parsed = load_from_path(mod_path, debug=True)
 
+    vars = [
+        {
+            "__tfmeta": {
+                "filename": "main.tf",
+                "label": "unknown",
+                "line_end": 53,
+                "line_start": 51,
+                "path": "variable.unknown",
+            },
+            "id": ANY,
+            "type": "set of string",
+        }
+    ]
+
     resource = {
         "__tfmeta": {
             "filename": "main.tf",
             "label": "some_resource",
-            "line_end": 41,
+            "line_end": 49,
             "line_start": 1,
             "path": ANY,
             "type": "resource",
@@ -363,6 +377,7 @@ def test_parse_dynamic_content(tmp_path):
             resource,
             resource,
         ],
+        "variable": vars,
     }
 
 

